File: sdks/dotnet/src/Backlex/BacklexException.cs
namespace Backlex;

/// <summary>
/// A non-2xx response from the backlex API, mirroring the TS SDK's BacklexError.
/// The API returns errors as <c>{ "error": { "code", "message", "details"? } }</c>;
/// callers branch on <see cref="Status"/> / <see cref="Code"/> rather than strings.
/// </summary>
public sealed class BacklexException : Exception
{
    /// <summary>HTTP status code.</summary>
    public int Status { get; }

    /// <summary>Machine-readable code ("VALIDATION", "UNAUTHORIZED", ...); "UNKNOWN" if absent.</summary>
    public string Code { get; }

    /// <summary>Optional structured details from the error envelope.</summary>
    public object? Details { get; }

    public BacklexException(int status, string code, string message, object? details)
        : base(message)
    {
        Status = status;
        Code = code;
        Details = details;
    }
}

File: sdks/java/src/main/java/com/backlex/BacklexException.java
package com.backlex;

/**
 * A non-2xx response from the backlex API (or a transport failure), mirroring the
 * TS SDK's BacklexError. The API returns errors as
 * {@code {"error": {"code","message","details"?}}}; callers branch on
 * {@link #status} / {@link #code} rather than parsing strings. Unchecked so the
 * call sites stay clean.
 */
public class BacklexException extends RuntimeException {

    /** HTTP status code (0 for transport/parse failures). */
    public final int status;

    /** Machine-readable code ("VALIDATION", "UNAUTHORIZED", ...); "UNKNOWN" if absent. */
    public final String code;

    /** Optional structured details from the error envelope. */
    public final Object details;

    public BacklexException(int status, String code, String message, Object details) {
        super(message);
        this.status = status;
        this.code = code;
        this.details = details;
    }
}
